pkg/updatestrategy/clc_update.go

```diff
@@ -0,0 +1,105 @@
+package updatestrategy
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando-incubator/cluster-lifecycle-manager/api"
+)
+
+type CLCUpdateStrategy struct {
+	nodePoolManager NodePoolManager
+	logger          *log.Entry
+	pollingInterval time.Duration
+}
+
+// NewCLCUpdateStrategy initializes a new CLCUpdateStrategy.
+func NewCLCUpdateStrategy(logger *log.Entry, nodePoolManager NodePoolManager, pollingInterval time.Duration) *CLCUpdateStrategy {
+	return &CLCUpdateStrategy{
+		nodePoolManager: nodePoolManager,
+		logger:          logger.WithField("strategy", "clc"),
+		pollingInterval: pollingInterval,
+	}
+}
+
+func (c *CLCUpdateStrategy) Update(ctx context.Context, nodePoolDesc *api.NodePool) error {
+	c.logger.Infof("Initializing update of node pool '%s'", nodePoolDesc.Name)
+
+	err := c.doUpdate(ctx, nodePoolDesc)
+	if err != nil {
+		if ctx.Err() != nil {
+			err := c.rollbackUpdate(nodePoolDesc)
```

Comment:
Is the intention that the CLC will handle this in the future? I understand why it's done this way, but it seems like the CLC should do this by itself at some point or is that not the intention?

Reply:
I don't really have any plans for this, we could move cluster-update blocking to the clusters as well I guess?